data-rights-protocol.md

```diff
@@ -288,20 +292,29 @@ Servers SHALL respond with HTTP 200 response codes when requests are processed s
+PIP servers MAY signal that an existing request will no longer be processed due to this error. PIP SHOULD move the request to a `denied`/`other` state and call the [Status Callback endpoint](#204-post-status_callback-data-rights-status-callback-endpoint) accordingly.
 
 Error codes are purposefully under-specified at the moment -- servers SHALL make a best effort to map to known 4XX and 5XX codes.
 
 Note that these error states only represent *request errors*; workflow errors SHOULD be specified in the request status fields.
 
 ### 3.07 API Authentication
 
-In short:
-- for v.0.5 we specify that client shared secrets will be used for authentication to all endpoints except the Data rights Discovery endpoint.
-- Participating parties will need to exchange shared secrets out of band for now
-  - the intention is to eventually leverage OAuth2 to secure these resources, either in concert with OIDC or out of band
-- Each party MUST include an HTTP `Authorization` header in each response containing the SHA-512 hash of their secret.
-- Requests which do not have an `Authorization` header MUST receive an `401` HTTP response.
+By enveloping the entire Data Rights Request inside of a JSON Web Token, it is possible to rely on the cryptographic signature of the message to authenticate it, rather than relying on `Authorization` headers or more complicated systems like `OAuth2`. 
+    
+In version 0.6 we continue to use JWT technology but in the future we will be evaluating a move to another cryptographic envelope library called [libsodium](https://doc.libsodium.org/) which eschews the URL-encoding and provides a much less error-prone API **without exchanging any pair-wise secrets**. The Data Rights Protocol organizing group will be responsible for managing a directory mapping `kid` to public API keys for participants in the network.
+
+0.6 should then be seen as a bridge-effort between the current API security paradigm and a final productionized version. Implementers are encouraged to review libsodium and the programming language bindings which are applicable to their technical stack and provide feedback to the Technical Working Group.
+
+We believe that providing these signed messages will ensure message integrity and prevent re-use or re-appropriate of requests: A data rights request represents a single action of one user acting against one covered business one time.
+
+Privacy Infrastructure Providers should take care to validate the message in this order:
+- That the signature validates to the key associated with the out of band Authorized Agent identity specified in `kid` URL parameter.
+- That the Authorized Agent attribute in the request matches the Authorized Agent identity outside the envelope
+- That they are the Covered Business specified inside the `aud` claim (this is very important to prevent request reuse)
```

Comment:
```suggestion
- That they are the Covered Business specified inside the `aud` claim (this is very important to prevent requests originally destined for one CB from being resent to other CBs)
```